browser/profiles/brave_profile_manager_unittest.cc

```diff
@@ -100,3 +151,220 @@ TEST_F(BraveProfileManagerTest, EnableMediaRouterOnRestartNonDefaultValue) {
+
+  if (IsAIChatAgentProfileFeatureEnabled()) {
+    // When feature is enabled, AI Chat Agent profile should NOT update last
+    // used time
+    EXPECT_EQ(ai_chat_entry->GetActiveTime(), ai_chat_time_before);
+  } else {
+    // When feature is disabled, it should behave like a regular profile
+    EXPECT_GE(ai_chat_entry->GetActiveTime(), ai_chat_time_before);
+  }
+}
+
+TEST_P(BraveProfileManagerAIAgentProfileTest, GetNumberOfProfiles) {
+  BraveProfileManager* profile_manager =
+      static_cast<BraveProfileManager*>(g_browser_process->profile_manager());
+  ASSERT_TRUE(profile_manager);
+
+  // Initially should have 0 profiles
+  EXPECT_EQ(0u, profile_manager->GetNumberOfProfiles());
+
+  // Add a regular profile
+  CreateRegularProfile("test_profile");
+  EXPECT_EQ(1u, profile_manager->GetNumberOfProfiles());
+
+  // Create an AI Chat Agent profile. We create it for both when the feature is
+  // enabled and when it is disabled because if the feature flag is disabled,
+  // the profile will still exist and be useable as a regular profile.
+  CreateAIChatAgentProfile();
+
+  if (IsAIChatAgentProfileFeatureEnabled()) {
+    // When feature is enabled, AI Chat Agent profile should NOT be counted
+    EXPECT_EQ(1u, profile_manager->GetNumberOfProfiles());
+  } else {
+    // When feature is disabled, it should be counted as a regular profile
+    EXPECT_EQ(2u, profile_manager->GetNumberOfProfiles());
+  }
+
+  // Add another regular profile
+  CreateRegularProfile("test_profile2");
+
+  if (IsAIChatAgentProfileFeatureEnabled()) {
+    EXPECT_EQ(2u, profile_manager->GetNumberOfProfiles());
+  } else {
+    EXPECT_EQ(3u, profile_manager->GetNumberOfProfiles());
+  }
+}
+
+TEST_P(BraveProfileManagerAIAgentProfileTest, GetLastOpenedProfiles) {
+  ProfileManager* profile_manager = g_browser_process->profile_manager();
+  ASSERT_TRUE(profile_manager);
+
+  // Create profiles
+  auto* regular_profile1 = CreateRegularProfile("regular1");
+  auto* regular_profile2 = CreateRegularProfile("regular2");
+  auto* ai_chat_profile = CreateAIChatAgentProfile();
+
+  base::FilePath regular1_path = regular_profile1->GetPath();
+  base::FilePath regular2_path = regular_profile2->GetPath();
+  base::FilePath ai_chat_path = ai_chat_profile->GetPath();
+
+  std::vector<Profile*> last_opened_profiles =
+      profile_manager->GetLastOpenedProfiles();
+  ASSERT_EQ(0U, last_opened_profiles.size());
+
+  // Simulate opening profiles by setting them as active in the same way
+  // as profile_manager_unittest.cc
+  Browser::CreateParams profile1_params(regular_profile1, true);
+  std::unique_ptr<Browser> browser1a(
+      CreateBrowserWithTestWindowForParams(profile1_params));
+
+  last_opened_profiles = profile_manager->GetLastOpenedProfiles();
+  ASSERT_EQ(1U, last_opened_profiles.size());
+  EXPECT_TRUE(profile_manager->has_updated_last_opened_profiles());
+  EXPECT_EQ(regular_profile1, last_opened_profiles[0]);
+
+  // And for ai chat profile
```

Comment:
can there be an edge case where all the regular profiles are deleted and we only have the aichat profile left?

Reply:
as discussed this applies when browsers are closed, not profiles deletes so I added a case for that